types/ntlm-client/index.d.ts

```diff
@@ -0,0 +1,32 @@
+// Type definitions for ntlm-client 0.1
+// Project: https://github.com/clncln1/node-ntlm-client
+// Definitions by: Manuel Borrajo <https://github.com/borrajo>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import "node";
+
+interface NtlmType2 {
+    flags: number;
+    encoding: string;
+    version: number;
+    challenge: Buffer;
+    targetName: string;
+    targetInfo: {
+        parsed: {
+            DOMAIN: string;
+            SERVER: string;
+            DNS: string;
+            FQDN: string;
+            PARENT_DNS: string;
+        },
+        buffer: Buffer;
+    };
+}
+
+declare namespace ntlm {
```

Comment:
Does this need to be namespaced?

Reply:
if you use 

```ts
import ntlm = require('ntlm-client');
```

in the test, you will see that you have to type
```ts
ntlm.ntlm.createType3Message
```
with the extra namespace layer.

Reply:
Do you need the ntlm types as well? You can export those as interfaces:
https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-d-ts.html

Reply:
That interface does not need to be exported, it is for internal use. The user has nothing to modify.